src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/SingleProjectTest.cs

```diff
@@ -11,13 +12,25 @@ namespace Xamarin.Android.Build.Tests
 	[Parallelizable (ParallelScope.Children)]
 	public partial class SingleProjectTest : BaseTest
 	{
+		static readonly object [] AndroidManifestPropertiesSource = new object [] {
+			new object [] {
+				/* versionName */ "2.1",
+				/* versionCode */ "42",
+				/* errorCode */   "",
+			},
+			new object [] {
+				/* versionName */ "1.0.0",
+				/* versionCode */ "1.0.0",
+				/* errorCode */   "XA0003",
+			},
```

Comment:
I need to add two test cases here:
* `3.1.3a1` - iOS number
* `6.0.100-preview.7` - semver

Just to see what happens, and if we need to do anything for these.

The iOS number comes from:
> While developing a new version of your app, you can include a suffix after the number that is being updated; for example 3.1.3a1. The character in the suffix represents the stage of development for the new version. For example, you can represent development, alpha, beta, and final candidate, by d, a, b, and fc. The final number in the suffix is the build version, which cannot be 0 and cannot exceed 255. When you release the new version of your app, remove the suffix.

Reply:
This one fails pretty badly...
```
~\android-toolchain\dotnet\sdk\6.0.100-rc.2.21456.2\NuGet.targets(130,5): '3.1.3a1' is not a valid version string. (Parameter 'value')
```

Reply:
I can't find anything sensible to do here, `$([System.Version]::Parse())` or `$([MSBuild]:VersionGreaterThanOrEquals())` throw during evaluation.

I think adding the test case is fine for now, and you would get the above error from `NuGet.targets`?

Reply:
Hmm 🤷 I'm not sure what the best option is. Especially since the error is in Nuget. Could we have a task that runs before `Restore` that checks and gives a more actionable "android"/"iOS" error message?

Reply:
Well, the same issue happens if you put `<Version>3.1.3a1</Version>` in a plain .NET class library.

So the error message needs to be improved in NuGet? So it helps all project types?